### DISCUSSION TITLE
Where does the generated code reside?

### DISCUSSION THREAD

**Discussion by indraneelr** [Budibase/budibase#4803] [Help]
Hi, I was trying to figure out where does the code generated by budibase reside ? 
That can be of great help when i want make some customizations and also in versioning.

**Comment by mjashanks** [comment]
Hi @indraneelr - the generated code is not really editable. If you we're to edit it, your changes would be overwritten on the next change - Budibase is just not built this way.

In the future, we plan to allow you to write your own front end components and backend datasources.

**Comment by indraneelr** [comment]
Thanks @mjashanks. What is the current mechanism for version control?

**Reply by mjashanks** [reply]
There isn't one at the moment :/

We are planning on adding this over the next 2-3 months
